// nitron/dist/chunk-JJ4NDXSO.js
#!/usr/bin/env node

// src/signer.ts
import { execFile } from "child_process";
import { access, mkdir } from "fs/promises";
import { join } from "path";
import { promisify } from "util";
import { fileURLToPath } from "url";
import { dirname } from "path";
import prompts from "prompts";
var execFileAsync = promisify(execFile);
var __filename = fileURLToPath(import.meta.url);
var __dirname = dirname(__filename);
function getSignerJarPath() {
  return join(__dirname, "..", "vendor", "uber-apk-signer.jar");
}
async function findJava() {
  const javaHomeRaw = process.env.JAVA_HOME;
  if (javaHomeRaw) {
    const javaHome = javaHomeRaw.trim();
    const ext = process.platform === "win32" ? ".exe" : "";
    const javaBin = join(javaHome, "bin", `java${ext}`);
    try {
      await execFileAsync(javaBin, ["-version"]);
      return javaBin;
    } catch (err) {
      console.error("JAVA_HOME check failed:", err);
    }
  }
  try {
    await execFileAsync("java", ["-version"]);
    return "java";
  } catch {
    throw new Error(
      "Java not found \u2014 Nitron needs Java Runtime 8+ for APK signing.\n  Install from: https://adoptium.net/\n  Or set JAVA_HOME to your Java installation."
    );
  }
}
async function findKeytool() {
  const javaHomeRaw = process.env.JAVA_HOME;
  if (javaHomeRaw) {
    const javaHome = javaHomeRaw.trim();
    const ext = process.platform === "win32" ? ".exe" : "";
    const keytoolBin = join(javaHome, "bin", `keytool${ext}`);
    try {
      await execFileAsync(keytoolBin, ["-help"]);
      return keytoolBin;
    } catch (err) {
    }
  }
  try {
    await execFileAsync("keytool", ["-help"]);
    return "keytool";
  } catch {
    throw new Error(
      "Keytool not found \u2014 Nitron needs the Java JDK (not just JRE) to generate a keystore.\n  Install the full JDK from: https://adoptium.net/\n  Or set JAVA_HOME to your JDK installation."
    );
  }
}
async function signApk(unsignedApkPath, outputDir, options) {
  const java = await findJava();
  const signerJar = getSignerJarPath();
  try {
    await access(signerJar);
  } catch {
    throw new Error(
      `uber-apk-signer.jar not found \u2014 the Nitron installation may be corrupted.
  Expected at: ${signerJar}`
    );
  }
  await mkdir(outputDir, { recursive: true });
  const args = [
    "-jar",
    signerJar,
    "--apks",
    unsignedApkPath,
    "--out",
    outputDir,
    "--allowResign"
  ];
  if (options?.release && options.projectDir) {
    const keystorePath = join(options.projectDir, "nitron-release.keystore");
    try {
      await access(keystorePath);
    } catch {
      throw new Error(
        'Release keystore not found!\nRun "npx nitron keystore" to generate a release keystore before building with --release.'
      );
    }
    const { password } = await prompts({
      type: "password",
      name: "password",
      message: "Enter release keystore password"
    });
    if (!password) {
      throw new Error("Password is required for release signing.");
    }
    args.push(
      "--ks",
      keystorePath,
      "--ksAlias",
      "release",
      "--ksPass",
      password,
      "--ksKeyPass",
      password
    );
  }
  try {
    const { stdout, stderr } = await execFileAsync(java, args, {
      timeout: 6e4
      // 60 second timeout
    });
    const { readdir } = await import("fs/promises");
    const files = await readdir(outputDir);
    const signedApk = files.find((f) => f.endsWith("-aligned-debugSigned.apk") || f.endsWith("-aligned-signed.apk"));
    if (!signedApk) {
      const anyApk = files.find((f) => f.endsWith(".apk"));
      if (anyApk) {
        return join(outputDir, anyApk);
      }
      throw new Error("Signing completed but no signed APK found in output");
    }
    return join(outputDir, signedApk);
  } catch (err) {
    if (err.code === "ENOENT") {
      throw new Error("Java not found during signing \u2014 ensure Java is installed and in PATH");
    }
    throw new Error(
      `APK signing failed: ${err.message}
  Make sure Java Runtime 8+ is installed.`
    );
  }
}

export {
  findKeytool,
  signApk
};
