// nitron/dist/chunk-J4YGGDBQ.js
#!/usr/bin/env node

// src/logger.ts
import pc from "picocolors";
var logger = {
  /**
   * Print an informational message.
   */
  info(msg) {
    console.log(`${pc.cyan("\u2139")} ${msg}`);
  },
  /**
   * Print a success message.
   */
  success(msg) {
    console.log(`${pc.green("\u2713")} ${msg}`);
  },
  /**
   * Print a warning (non-fatal).
   */
  warn(msg) {
    console.log(`${pc.yellow("\u26A0")} ${msg}`);
  },
  /**
   * Print an error (fatal).
   */
  error(msg) {
    console.error(`${pc.red("\u2717")} ${msg}`);
  },
  /**
   * Print a build step indicator.
   * Example: [1/8] Reading configuration...
   */
  step(step, total, msg) {
    const badge = pc.dim(`[${step}/${total}]`);
    console.log(`${badge} ${msg}`);
  },
  /**
   * Print a blank line for visual spacing.
   */
  blank() {
    console.log();
  },
  /**
   * Print a section header with underline.
   */
  header(msg) {
    console.log();
    console.log(pc.bold(pc.cyan(`  ${msg}`)));
    console.log(pc.dim(`  ${"\u2500".repeat(40)}`));
  },
  /**
   * Print a config key-value pair with aligned formatting.
   */
  config(key, value) {
    const paddedKey = key.padEnd(14);
    console.log(`  ${pc.dim(paddedKey)} ${value}`);
  },
  /**
   * Print the Nitron banner / logo.
   */
  banner() {
    console.log();
    console.log(pc.bold(pc.cyan("  \u26A1 Nitron")));
    console.log(pc.dim("  HTML/CSS/JS \u2192 Android APK"));
    console.log();
  },
  /**
   * Print a final build summary line.
   */
  summary(outputPath, size, duration) {
    console.log();
    console.log(
      `${pc.green("\u2713")} ${pc.bold("Built successfully")} \u2192 ${pc.cyan(outputPath)} ${pc.dim(`(${size}) in ${duration}`)}`
    );
    console.log();
  }
};

export {
  logger
};
