// nitron/dist/pwa-OWYG7CMO.js
#!/usr/bin/env node
import {
  logger
} from "./chunk-J4YGGDBQ.js";

// src/pwa.ts
import { join, relative } from "path";
import { mkdir, readdir, copyFile, readFile, writeFile, rm } from "fs/promises";
var EXCLUDED = /* @__PURE__ */ new Set([
  "app.js",
  "package.json",
  "package-lock.json",
  "node_modules",
  "dist",
  ".git",
  ".gitignore",
  ".DS_Store",
  "Thumbs.db"
]);
async function buildPwa(projectDir, config) {
  const pwaDir = join(projectDir, "dist", "pwa");
  await rm(pwaDir, { recursive: true, force: true });
  await mkdir(pwaDir, { recursive: true });
  logger.step(1, 4, "Copying web assets...");
  const copiedFiles = [];
  async function copyRecursive(srcDir, destDir) {
    const entries = await readdir(srcDir, { withFileTypes: true });
    for (const entry of entries) {
      if (EXCLUDED.has(entry.name)) continue;
      const srcPath = join(srcDir, entry.name);
      const destPath = join(destDir, entry.name);
      if (entry.isDirectory()) {
        await mkdir(destPath, { recursive: true });
        await copyRecursive(srcPath, destPath);
      } else if (entry.isFile()) {
        await copyFile(srcPath, destPath);
        const relPath = relative(pwaDir, destPath).replace(/\\/g, "/");
        copiedFiles.push(relPath);
      }
    }
  }
  await copyRecursive(projectDir, pwaDir);
  logger.success(`Copied ${copiedFiles.length} files`);
  logger.step(2, 4, "Generating manifest.json...");
  const manifest = {
    name: config.name,
    short_name: config.name,
    start_url: `./${config.entry}`,
    display: "standalone",
    orientation: config.orientation === "portrait" ? "portrait" : config.orientation === "landscape" ? "landscape" : "any",
    background_color: "#ffffff",
    theme_color: "#0070f3",
    icons: [
      {
        src: config.icon === "default" || !config.icon ? "icon.png" : config.icon,
        sizes: "192x192 512x512",
        type: "image/png",
        purpose: "any maskable"
      }
    ]
  };
  await writeFile(join(pwaDir, "manifest.json"), JSON.stringify(manifest, null, 2));
  if (config.icon === "default" && !copiedFiles.includes("icon.png")) {
    logger.warn("Default icon requested but icon.png not found. Add an icon.png for better PWA experience.");
  }
  logger.step(3, 4, "Generating service-worker.js...");
  const cacheName = `${config.packageId}-v${config.version}`;
  const cacheList = [...copiedFiles.map((f) => `./${f}`), "./manifest.json"];
  const swContent = `
const CACHE_NAME = '${cacheName}';
const ASSETS = ${JSON.stringify(cacheList, null, 2)};

// Install Event: Cache all assets
self.addEventListener('install', event => {
  event.waitUntil(
    caches.open(CACHE_NAME)
      .then(cache => {
        console.log('[Service Worker] Caching all assets');
        return cache.addAll(ASSETS);
      })
      .then(() => self.skipWaiting())
  );
});

// Activate Event: Cleanup old caches
self.addEventListener('activate', event => {
  event.waitUntil(
    caches.keys().then(cacheNames => {
      return Promise.all(
        cacheNames.map(cache => {
          if (cache !== CACHE_NAME) {
            console.log('[Service Worker] Clearing old cache:', cache);
            return caches.delete(cache);
          }
        })
      );
    }).then(() => self.clients.claim())
  );
});

// Fetch Event: Cache First Strategy
self.addEventListener('fetch', event => {
  event.respondWith(
    caches.match(event.request)
      .then(response => {
        if (response) {
          return response; // Return from cache
        }
        return fetch(event.request); // Network fallback
      })
  );
});
`;
  await writeFile(join(pwaDir, "service-worker.js"), swContent);
  logger.step(4, 4, "Injecting PWA tags into HTML...");
  const indexPath = join(pwaDir, config.entry);
  try {
    let html = await readFile(indexPath, "utf-8");
    if (!html.includes("manifest.json")) {
      const manifestTag = '\n  <link rel="manifest" href="manifest.json">';
      html = html.replace("</head>", `${manifestTag}
</head>`);
    }
    if (!html.includes("serviceWorker")) {
      const swTag = `
  <script>
    if ('serviceWorker' in navigator) {
      window.addEventListener('load', () => {
        navigator.serviceWorker.register('./service-worker.js')
          .then(reg => console.log('SW registered:', reg.scope))
          .catch(err => console.error('SW failed:', err));
      });
    }
  </script>
`;
      html = html.replace("</body>", `${swTag}
</body>`);
    }
    await writeFile(indexPath, html);
  } catch (err) {
    logger.warn(`Could not update ${config.entry}: ${err.message}`);
  }
  return { outDir: pwaDir, filesCount: copiedFiles.length + 2 };
}
export {
  buildPwa
};
